<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Use of IlluminaHumanMethylationEPICv2anno.20a1.hg38 • IlluminaHumanMethylationEPICv2anno.20a1.hg38</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Use of IlluminaHumanMethylationEPICv2anno.20a1.hg38">
<meta property="og:description" content="IlluminaHumanMethylationEPICv2anno.20a1.hg38">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">IlluminaHumanMethylationEPICv2anno.20a1.hg38</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/IlluminaHumanMethylationEPICv2anno.20a1.hg38.html">Use of IlluminaHumanMethylationEPICv2anno.20a1.hg38</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Use of IlluminaHumanMethylationEPICv2anno.20a1.hg38</h1>
                        <h4 data-toc-skip class="author">Zuguang Gu (<a href="mailto:z.gu@dkfz.de" class="email">z.gu@dkfz.de</a>)</h4>
            
            <h4 data-toc-skip class="date">2024-03-14</h4>
      
      
      <div class="hidden name"><code>IlluminaHumanMethylationEPICv2anno.20a1.hg38.Rmd</code></div>

    </div>

    
    
<p>This package provides annotations for Illumina methylation EPIC array v2.0. The data is based on the file <a href="https://support.illumina.com/content/dam/illumina-support/documents/downloads/productfiles/methylationepic/MethylationEPIC%20v2%20Files.zip" class="external-link uri">https://support.illumina.com/content/dam/illumina-support/documents/downloads/productfiles/methylationepic/MethylationEPIC%20v2%20Files.zip</a> from <a href="https://support.illumina.com/array/array_kits/infinium-methylationepic-beadchip-kit/downloads.html" class="external-link uri">https://support.illumina.com/array/array_kits/infinium-methylationepic-beadchip-kit/downloads.html</a>.</p>
<p>When using with the <strong>minfi</strong> package, you can manually set the “annotation” element by providing the suffix (removing the string “IlluminaHumanMethylationEPICv2anno.”):</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">RGset</span> <span class="op">=</span> <span class="fu">read.metharray.exp</span><span class="op">(</span><span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># explained in the IlluminaHumanMethylationEPICv2manifest package</span></span>
<span><span class="fu">annotation</span><span class="op">(</span><span class="va">RGset</span><span class="op">)</span><span class="op">[</span><span class="st">"array"</span><span class="op">]</span> <span class="op">=</span> <span class="st">"IlluminaHumanMethylationEPICv2"</span></span>
<span></span>
<span><span class="fu">annotation</span><span class="op">(</span><span class="va">RGset</span><span class="op">)</span><span class="op">[</span><span class="st">"annotation"</span><span class="op">]</span> <span class="op">=</span> <span class="st">"20a1.hg38"</span></span></code></pre></div>
<div class="section level2">
<h2 id="compare-epic-array-v1-and-v2-probes">Compare EPIC array v1 and v2 probes<a class="anchor" aria-label="anchor" href="#compare-epic-array-v1-and-v2-probes"></a>
</h2>
<p>We compare CpG annotation in the package <strong>IlluminaHumanMethylationEPICv2anno.20a1.hg38</strong> and the package <strong>IlluminaHumanMethylationEPICanno.ilm10b4.hg19</strong>.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://www.illumina.com/products/by-type/microarray-kits/infinium-methylation-epic.html" class="external-link">IlluminaHumanMethylationEPICv2anno.20a1.hg38</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bitbucket.com/kasperdanielhansen/Illumina_EPIC" class="external-link">IlluminaHumanMethylationEPICanno.ilm10b4.hg19</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">cgi1</span> <span class="op">=</span> <span class="fu">IlluminaHumanMethylationEPICanno.ilm10b4.hg19</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/IlluminaHumanMethylationEPICanno.ilm10b4.hg19/man/IlluminaHumanMethylationEPICanno.ilm10b4.hg19.html" class="external-link">Islands.UCSC</a></span></span>
<span><span class="va">cgi2</span> <span class="op">=</span> <span class="fu">IlluminaHumanMethylationEPICv2anno.20a1.hg38</span><span class="fu">::</span><span class="va"><a href="../reference/IlluminaHumanMethylationEPICv2anno.20a1.hg38.html">Islands.UCSC</a></span></span></code></pre></div>
<p>Following code shows the number of probes in each CpG feature. Note in <strong>IlluminaHumanMethylationEPICanno.ilm10b4.hg19</strong>, CGI shores and shelves are additionally classified as “N_Shore”/“S_Shore” and “N_Shelf”/“S_Shelf”, while they are simply “Shore” and “Shelf” in <strong>IlluminaHumanMethylationEPICv2anno.20a1.hg38</strong>.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">t1</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"(N|S)_"</span>, <span class="st">""</span>, <span class="va">cgi1</span><span class="op">$</span><span class="va">Relation_to_Island</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">t1</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">##  Island OpenSea   Shelf   Shore </span></span>
<span><span class="co">##  161441  488181   61691  154546</span></span></code></pre>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">t2</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">cgi2</span><span class="op">$</span><span class="va">Relation_to_Island</span><span class="op">)</span></span>
<span><span class="va">t2</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">##  Island OpenSea   Shelf   Shore </span></span>
<span><span class="co">##  150679  560737   60836  157823</span></span></code></pre>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">t2</span> <span class="op">-</span> <span class="va">t1</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">##  Island OpenSea   Shelf   Shore </span></span>
<span><span class="co">##  -10762   72556    -855    3277</span></span></code></pre>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="op">(</span><span class="va">t2</span> <span class="op">-</span> <span class="va">t1</span><span class="op">)</span><span class="op">/</span><span class="va">t1</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">##      Island     OpenSea       Shelf       Shore </span></span>
<span><span class="co">## -0.06666212  0.14862520 -0.01385940  0.02120404</span></span></code></pre>
<p>We can see there are more new probes in the CpG seas.</p>
</div>
<div class="section level2">
<h2 id="probe-ids-are-coded-differently-in-epic-v1-and-epic-v2-packages">Probe IDs are coded differently in EPIC v1 and EPIC v2 packages<a class="anchor" aria-label="anchor" href="#probe-ids-are-coded-differently-in-epic-v1-and-epic-v2-packages"></a>
</h2>
<p>In <strong>IlluminaHumanMethylationEPICanno.ilm10b4.hg19</strong> and other related EPIC-v1 packages, probe IDs (e.g. in the format of “cg18478105”) are unique in the array. But in the packages related to EPIC-v2, probe IDs may be duplicated. Thus, we use the “illumina_ID” (column “IlmnID” in the manifest file, <a href="https://knowledge.illumina.com/microarray/general/microarray-general-reference_material-list/000001568" class="external-link uri">https://knowledge.illumina.com/microarray/general/microarray-general-reference_material-list/000001568</a>) as the ID type for probes in v2-packages. The duplicated probes have the same probe sequence, but locate randomly on the array. The illumina ID is a combination of probe ID and a “duplication ID”.</p>
<p>Let’s take <code>cgi1</code> and <code>cgi2</code> as an example:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">cgi1</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 865859      2</span></span></code></pre>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">cgi1</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## DataFrame with 6 rows and 2 columns</span></span>
<span><span class="co">##                      Islands_Name Relation_to_Island</span></span>
<span><span class="co">##                       &lt;character&gt;        &lt;character&gt;</span></span>
<span><span class="co">## cg18478105 chr20:61846843-61848..             Island</span></span>
<span><span class="co">## cg09835024 chrX:24072558-24073135             Island</span></span>
<span><span class="co">## cg14361672 chr9:131464843-13146..            N_Shore</span></span>
<span><span class="co">## cg01763666                                   OpenSea</span></span>
<span><span class="co">## cg12950382                                   OpenSea</span></span>
<span><span class="co">## cg02115394 chr13:115000148-1150..             Island</span></span></code></pre>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">cgi2</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 930075      2</span></span></code></pre>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">cgi2</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## DataFrame with 6 rows and 2 columns</span></span>
<span><span class="co">##                           Islands_Name Relation_to_Island</span></span>
<span><span class="co">##                            &lt;character&gt;        &lt;character&gt;</span></span>
<span><span class="co">## cg25324105_BC11 chr19:37691892-37692..             Island</span></span>
<span><span class="co">## cg25383568_TC11 chr19:38726890-38727..             Island</span></span>
<span><span class="co">## cg25455143_BC11 chr19:1591428-159162..             Island</span></span>
<span><span class="co">## cg25459778_BC11  chr16:1610053-1615094             Island</span></span>
<span><span class="co">## cg25487775_BC11 chr2:161237949-16123..              Shore</span></span>
<span><span class="co">## cg25595446_BC11 chr19:49676753-49677..             Island</span></span></code></pre>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/any.html" class="external-link">any</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/duplicated.html" class="external-link">duplicated</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">cgi1</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] FALSE</span></span></code></pre>
<p>Let’s check how many probes have duplicated probe IDs. First remove the suffix to only keep the probe IDs.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">illumina_ID</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">cgi2</span><span class="op">)</span></span>
<span><span class="va">probe_ID</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"_.*$"</span>, <span class="st">""</span>, <span class="va">illumina_ID</span><span class="op">)</span></span></code></pre></div>
<p>Check the duplication of probe IDs:</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tb</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">probe_ID</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">tb</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## tb</span></span>
<span><span class="co">##      1      2      3      4      5      6     10 </span></span>
<span><span class="co">## 918546   4174    942     23     47      3      1</span></span></code></pre>
<p>We can see in the most extreme case, a probe ID is repeated 10 times in the array. Let’s check what it is:</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tb</span><span class="op">[</span><span class="va">tb</span> <span class="op">==</span> <span class="fl">10</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## cg06373096 </span></span>
<span><span class="co">##         10</span></span></code></pre>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">illumina_ID</span><span class="op">[</span><span class="va">probe_ID</span> <span class="op">==</span> <span class="st">"cg06373096"</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">##  [1] "cg06373096_TC11"  "cg06373096_TC12"  "cg06373096_TC13"  "cg06373096_TC14" </span></span>
<span><span class="co">##  [5] "cg06373096_TC15"  "cg06373096_TC16"  "cg06373096_TC17"  "cg06373096_TC18" </span></span>
<span><span class="co">##  [9] "cg06373096_TC19"  "cg06373096_TC110"</span></span></code></pre>
<p>But the locations of these 10 probes are the same:</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cgi2</span><span class="op">[</span><span class="va">probe_ID</span> <span class="op">==</span> <span class="st">"cg06373096"</span>, <span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## DataFrame with 10 rows and 2 columns</span></span>
<span><span class="co">##                            Islands_Name Relation_to_Island</span></span>
<span><span class="co">##                             &lt;character&gt;        &lt;character&gt;</span></span>
<span><span class="co">## cg06373096_TC11  chr19:12880876-12881..              Shore</span></span>
<span><span class="co">## cg06373096_TC12  chr19:12880876-12881..              Shore</span></span>
<span><span class="co">## cg06373096_TC13  chr19:12880876-12881..              Shore</span></span>
<span><span class="co">## cg06373096_TC14  chr19:12880876-12881..              Shore</span></span>
<span><span class="co">## cg06373096_TC15  chr19:12880876-12881..              Shore</span></span>
<span><span class="co">## cg06373096_TC16  chr19:12880876-12881..              Shore</span></span>
<span><span class="co">## cg06373096_TC17  chr19:12880876-12881..              Shore</span></span>
<span><span class="co">## cg06373096_TC18  chr19:12880876-12881..              Shore</span></span>
<span><span class="co">## cg06373096_TC19  chr19:12880876-12881..              Shore</span></span>
<span><span class="co">## cg06373096_TC110 chr19:12880876-12881..              Shore</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="change-illumina-ids-to-probe-ids">Change illumina IDs to probe IDs<a class="anchor" aria-label="anchor" href="#change-illumina-ids-to-probe-ids"></a>
</h2>
<p>If you think illumina IDs as “probe IDs” as in the v1 packages, you can basically do all the same analyses without worrying aboutn the format of the probe IDs. But if you really need the original probe IDs, you can use the <code><a href="../reference/aggregate_to_probes.html">aggregate_to_probes()</a></code> function. It works for both annotation data frames in <strong>IlluminaHumanMethylationEPICv2anno.20a1.hg38</strong> and the beta matrix normalized by the <strong>minfi</strong> package.</p>
<p>The following example reads raw data from one sample and performs preprocessing with <strong>minfi</strong>. <code><a href="../reference/aggregate_to_probes.html">aggregate_to_probes()</a></code> calculates mean value for duplicated probes.</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tempdir</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">datadir</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">tempdir</span>, <span class="st">"/206891110001"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/files2.html" class="external-link">dir.create</a></span><span class="op">(</span><span class="va">datadir</span>, showWarnings <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">url</span> <span class="op">=</span> <span class="st">"https://github.com/jokergoo/IlluminaHumanMethylationEPICv2manifest/files/11008723/206891110001_R01C01.zip"</span></span>
<span><span class="va">local</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">tempdir</span>, <span class="st">"/206891110001_R01C01.zip"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span><span class="va">url</span>, dest <span class="op">=</span> <span class="va">local</span>, quiet <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/unzip.html" class="external-link">unzip</a></span><span class="op">(</span><span class="va">local</span>, exdir <span class="op">=</span> <span class="va">datadir</span><span class="op">)</span></span>
<span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/hansenlab/minfi" class="external-link">minfi</a></span><span class="op">)</span></span>
<span><span class="va">RGset</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/minfi/man/read.metharray.exp.html" class="external-link">read.metharray.exp</a></span><span class="op">(</span><span class="va">datadir</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/annotation.html" class="external-link">annotation</a></span><span class="op">(</span><span class="va">RGset</span><span class="op">)</span><span class="op">[</span><span class="st">"array"</span><span class="op">]</span> <span class="op">=</span> <span class="st">"IlluminaHumanMethylationEPICv2"</span></span>
<span><span class="va">obj</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/minfi/man/preprocessRaw.html" class="external-link">preprocessRaw</a></span><span class="op">(</span><span class="va">RGset</span><span class="op">)</span></span>
<span><span class="co"># there can be more intermediate steps ...</span></span>
<span><span class="va">beta</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/minfi/man/MethylSet-class.html" class="external-link">getBeta</a></span><span class="op">(</span><span class="va">obj</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">beta</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 936990      1</span></span></code></pre>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">beta</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                 206891110001_R01C01</span></span>
<span><span class="co">## cg25324105_BC11          0.86065794</span></span>
<span><span class="co">## cg25383568_TC11          0.93799357</span></span>
<span><span class="co">## cg25455143_BC11          0.02141582</span></span>
<span><span class="co">## cg25459778_BC11          0.01751512</span></span>
<span><span class="co">## cg25487775_BC11          0.94253932</span></span>
<span><span class="co">## cg25595446_BC11          0.02561140</span></span></code></pre>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">beta2</span> <span class="op">=</span> <span class="fu"><a href="../reference/aggregate_to_probes.html">aggregate_to_probes</a></span><span class="op">(</span><span class="va">beta</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">beta2</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 930596      1</span></span></code></pre>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">beta2</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##            206891110001_R01C01</span></span>
<span><span class="co">## cg25324105          0.86065794</span></span>
<span><span class="co">## cg25383568          0.93799357</span></span>
<span><span class="co">## cg25455143          0.02141582</span></span>
<span><span class="co">## cg25459778          0.01751512</span></span>
<span><span class="co">## cg25487775          0.94253932</span></span>
<span><span class="co">## cg25595446          0.02561140</span></span></code></pre>
<p><code><a href="../reference/aggregate_to_probes.html">aggregate_to_probes()</a></code> can also be applied to the annotation data frames in <strong>IlluminaHumanMethylationEPICv2anno.20a1.hg38</strong>.</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="../reference/aggregate_to_probes.html">aggregate_to_probes</a></span><span class="op">(</span><span class="va">cgi2</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## DataFrame with 6 rows and 2 columns</span></span>
<span><span class="co">##                      Islands_Name Relation_to_Island</span></span>
<span><span class="co">##                       &lt;character&gt;        &lt;character&gt;</span></span>
<span><span class="co">## cg25324105 chr19:37691892-37692..             Island</span></span>
<span><span class="co">## cg25383568 chr19:38726890-38727..             Island</span></span>
<span><span class="co">## cg25455143 chr19:1591428-159162..             Island</span></span>
<span><span class="co">## cg25459778  chr16:1610053-1615094             Island</span></span>
<span><span class="co">## cg25487775 chr2:161237949-16123..              Shore</span></span>
<span><span class="co">## cg25595446 chr19:49676753-49677..             Island</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<pre><code><span><span class="co">## R version 4.3.1 (2023-06-16)</span></span>
<span><span class="co">## Platform: x86_64-apple-darwin20 (64-bit)</span></span>
<span><span class="co">## Running under: macOS Ventura 13.2.1</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/lib/libRblas.0.dylib </span></span>
<span><span class="co">## LAPACK: /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">## [1] C/UTF-8/C/C/C/C</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: Europe/Berlin</span></span>
<span><span class="co">## tzcode source: internal</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] parallel  stats4    stats     graphics  grDevices utils     datasets </span></span>
<span><span class="co">## [8] methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] IlluminaHumanMethylationEPICv2manifest_0.99.2      </span></span>
<span><span class="co">##  [2] IlluminaHumanMethylationEPICanno.ilm10b4.hg19_0.6.0</span></span>
<span><span class="co">##  [3] IlluminaHumanMethylationEPICv2anno.20a1.hg38_0.99.1</span></span>
<span><span class="co">##  [4] minfi_1.46.0                                       </span></span>
<span><span class="co">##  [5] bumphunter_1.42.0                                  </span></span>
<span><span class="co">##  [6] locfit_1.5-9.8                                     </span></span>
<span><span class="co">##  [7] iterators_1.0.14                                   </span></span>
<span><span class="co">##  [8] foreach_1.5.2                                      </span></span>
<span><span class="co">##  [9] Biostrings_2.68.1                                  </span></span>
<span><span class="co">## [10] XVector_0.40.0                                     </span></span>
<span><span class="co">## [11] SummarizedExperiment_1.30.2                        </span></span>
<span><span class="co">## [12] Biobase_2.60.0                                     </span></span>
<span><span class="co">## [13] MatrixGenerics_1.12.3                              </span></span>
<span><span class="co">## [14] matrixStats_1.2.0                                  </span></span>
<span><span class="co">## [15] GenomicRanges_1.52.1                               </span></span>
<span><span class="co">## [16] GenomeInfoDb_1.36.4                                </span></span>
<span><span class="co">## [17] IRanges_2.36.0                                     </span></span>
<span><span class="co">## [18] S4Vectors_0.40.2                                   </span></span>
<span><span class="co">## [19] BiocGenerics_0.48.1                                </span></span>
<span><span class="co">## [20] knitr_1.44                                         </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] RColorBrewer_1.1-3        jsonlite_1.8.8           </span></span>
<span><span class="co">##   [3] magrittr_2.0.3            GenomicFeatures_1.52.2   </span></span>
<span><span class="co">##   [5] rmarkdown_2.25            fs_1.6.3                 </span></span>
<span><span class="co">##   [7] BiocIO_1.10.0             zlibbioc_1.46.0          </span></span>
<span><span class="co">##   [9] ragg_1.2.6                vctrs_0.6.4              </span></span>
<span><span class="co">##  [11] multtest_2.56.0           memoise_2.0.1            </span></span>
<span><span class="co">##  [13] Rsamtools_2.16.0          DelayedMatrixStats_1.22.6</span></span>
<span><span class="co">##  [15] RCurl_1.98-1.12           askpass_1.2.0            </span></span>
<span><span class="co">##  [17] htmltools_0.5.7           S4Arrays_1.0.6           </span></span>
<span><span class="co">##  [19] progress_1.2.2            curl_5.1.0               </span></span>
<span><span class="co">##  [21] Rhdf5lib_1.22.1           rhdf5_2.44.0             </span></span>
<span><span class="co">##  [23] sass_0.4.8                nor1mix_1.3-2            </span></span>
<span><span class="co">##  [25] bslib_0.6.1               desc_1.4.2               </span></span>
<span><span class="co">##  [27] plyr_1.8.9                cachem_1.0.8             </span></span>
<span><span class="co">##  [29] GenomicAlignments_1.36.0  lifecycle_1.0.4          </span></span>
<span><span class="co">##  [31] pkgconfig_2.0.3           Matrix_1.6-1.1           </span></span>
<span><span class="co">##  [33] R6_2.5.1                  fastmap_1.1.1            </span></span>
<span><span class="co">##  [35] GenomeInfoDbData_1.2.10   digest_0.6.33            </span></span>
<span><span class="co">##  [37] siggenes_1.74.0           reshape_0.8.9            </span></span>
<span><span class="co">##  [39] AnnotationDbi_1.62.2      rprojroot_2.0.3          </span></span>
<span><span class="co">##  [41] textshaping_0.3.7         RSQLite_2.3.1            </span></span>
<span><span class="co">##  [43] base64_2.0.1              filelock_1.0.2           </span></span>
<span><span class="co">##  [45] fansi_1.0.5               httr_1.4.7               </span></span>
<span><span class="co">##  [47] abind_1.4-5               compiler_4.3.1           </span></span>
<span><span class="co">##  [49] beanplot_1.3.1            rngtools_1.5.2           </span></span>
<span><span class="co">##  [51] bit64_4.0.5               BiocParallel_1.34.2      </span></span>
<span><span class="co">##  [53] DBI_1.1.3                 HDF5Array_1.28.1         </span></span>
<span><span class="co">##  [55] biomaRt_2.56.1            MASS_7.3-60              </span></span>
<span><span class="co">##  [57] openssl_2.1.1             rappdirs_0.3.3           </span></span>
<span><span class="co">##  [59] DelayedArray_0.26.7       rjson_0.2.21             </span></span>
<span><span class="co">##  [61] tools_4.3.1               quadprog_1.5-8           </span></span>
<span><span class="co">##  [63] glue_1.6.2                restfulr_0.0.15          </span></span>
<span><span class="co">##  [65] nlme_3.1-163              rhdf5filters_1.12.1      </span></span>
<span><span class="co">##  [67] grid_4.3.1                generics_0.1.3           </span></span>
<span><span class="co">##  [69] tzdb_0.4.0                preprocessCore_1.62.1    </span></span>
<span><span class="co">##  [71] tidyr_1.3.0               data.table_1.14.8        </span></span>
<span><span class="co">##  [73] hms_1.1.3                 xml2_1.3.6               </span></span>
<span><span class="co">##  [75] utf8_1.2.3                pillar_1.9.0             </span></span>
<span><span class="co">##  [77] stringr_1.5.0             limma_3.56.2             </span></span>
<span><span class="co">##  [79] genefilter_1.82.1         splines_4.3.1            </span></span>
<span><span class="co">##  [81] dplyr_1.1.3               BiocFileCache_2.8.0      </span></span>
<span><span class="co">##  [83] lattice_0.21-9            survival_3.5-8           </span></span>
<span><span class="co">##  [85] rtracklayer_1.60.1        bit_4.0.5                </span></span>
<span><span class="co">##  [87] GEOquery_2.68.0           annotate_1.78.0          </span></span>
<span><span class="co">##  [89] tidyselect_1.2.0          xfun_0.40                </span></span>
<span><span class="co">##  [91] scrime_1.3.5              stringi_1.7.12           </span></span>
<span><span class="co">##  [93] yaml_2.3.7                evaluate_0.22            </span></span>
<span><span class="co">##  [95] codetools_0.2-19          tibble_3.2.1             </span></span>
<span><span class="co">##  [97] cli_3.6.2                 xtable_1.8-4             </span></span>
<span><span class="co">##  [99] systemfonts_1.0.5         jquerylib_0.1.4          </span></span>
<span><span class="co">## [101] Rcpp_1.0.11               dbplyr_2.3.4             </span></span>
<span><span class="co">## [103] png_0.1-8                 XML_3.99-0.14            </span></span>
<span><span class="co">## [105] readr_2.1.4               pkgdown_2.0.7            </span></span>
<span><span class="co">## [107] blob_1.2.4                prettyunits_1.2.0        </span></span>
<span><span class="co">## [109] mclust_6.0.0              doRNG_1.8.6              </span></span>
<span><span class="co">## [111] sparseMatrixStats_1.12.2  bitops_1.0-7             </span></span>
<span><span class="co">## [113] illuminaio_0.42.0         purrr_1.0.2              </span></span>
<span><span class="co">## [115] crayon_1.5.2              rlang_1.1.2              </span></span>
<span><span class="co">## [117] KEGGREST_1.40.1</span></span></code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Zuguang Gu.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
